### ISSUE TITLE
Ruby 2.4 error: Warning is not a class (TypeError)

### ISSUE THREAD

**Issue by esambo** [djberg96/attempt#1]
Ruby 2.4 now has it's own [Warning](https://ruby-doc.org/core-2.4.0/Warning.html) class which causes the error:
```
Warning is not a class (TypeError)
~/.gem/ruby/2.4.0/gems/structured_warnings-0.2.0/lib/structured_warnings/warning.rb:11:in '<top (required)>'
~/.gem/ruby/2.4.0/gems/structured_warnings-0.2.0/lib/structured_warnings.rb:4:in '<top (required)>'
~/.gem/ruby/2.4.0/gems/attempt-0.2.1/lib/attempt.rb:2:in '<top (required)>'
```
This error is due to the dependency on [structured_warnings (0.2.0)](rubygems.org/gems/structured_warnings) which could be fixed with something like [structured_warnings#7](http://github.com/schmidt/structured_warnings/pull/7)

**Comment by esambo** [comment]
structured_warnings just released version 0.3.0 which is now compatible with Ruby 2.4.x
See the [compatibility note](https://github.com/schmidt/structured_warnings#compatibility) for how to upgrade

**Comment by djberg96** [comment]
I pushed out 0.3.0 which updates the constant and dependency.
